roles/openshift_master/tasks/main.yml

```diff
@@ -1,13 +1,45 @@
 ---
-# TODO: add validation for openshift_master_identity_providers
 # TODO: add ability to configure certificates given either a local file to
 #       point to or certificate contents, set in default cert locations.
 
-- assert:
-    that:
-    - openshift_master_oauth_grant_method in openshift_master_valid_grant_methods
-  when: openshift_master_oauth_grant_method is defined
+# Authentication Variable Validation
+# TODO: validate the different identity provider kinds as well
+- fail:
+    msg: >
+      Invalid OAuth grant method: {{ openshift_master_oauth_grant_method }}
+  when: openshift_master_oauth_grant_method is defined and openshift_master_oauth_grant_method not in openshift_master_valid_grant_methods
+- fail:
+    msg: >
+      Missing required key {{ item.0 }} for identity provider {{ item.1.name }}
+  when: item.0 not in item.1
+  with_nested:
+  - - challenge
+    - login
+    - kind
+    - name
+  - openshift_master_identity_providers
+- fail:
+    msg: >
+      Invalid mapping method: {{ item.mappingMethod }} specified for
+      identity provider: {{ item.name }}
+  when: "openshift_master_identity_providers is defined and 'mappingMethod' in item and item.mappingMethod not in openshift_master_valid_mapping_methods"
+  with_items: openshift_master_identity_providers
+- fail:
+    msg: >
+      More than one identity provider configured as a login provider
+  when: "openshift_master_identity_providers is defined and openshift_master_identity_providers | oo_collect('login', {'login': 'true'}) | length > 1"
+- fail:
+    msg: >
+      More than one identity provider configured with the same name
+  when: "openshift_master_identity_providers is defined and openshift_master_identity_providers | oo_collect('name') | unique | length < openshift_master_identity_providers | length"
+- fail:
```

Comment:
We don't support arbitrary snippets currently, though maybe that would be easier and clean up the ugly template. However then we would lose the ability to do any validation before the master service fails to start...  Trade off between limited support and simple implementation.